### PR TITLE
farge: init at 1.0.8

### PR DESCRIPTION
**Pull request by loafofpiecrust**
###### Motivation for this change
I like using [farge](https://github.com/sdushantha/farge) for picking colors, since there doesn't seem to be a solid equivalent on Wayland.
I'm also excited to make my first contribution to `nixpkgs`.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux) (only tested in Wayland)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) (N/A)
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by laikq** [comment]
Hello :) I'm not a nixpkgs guru. But I would add two things:
1) Adding a package to top-level/all-packages isn't worth a separate commit, this change should go into `farge: init at 1.0.8`. I think instead it would make more sense to have two commits, `maintainers: add loafofpiecrust` (where you only add yourself to the maintainers list) and `farge: init at 1.0.8`.
2) This packages doesn't work for me on X11, it seems it needs `colorpicker` for that.

File: pkgs/applications/graphics/farge/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ config, lib, pkgs, stdenv, fetchFromGitHub, waylandSupport ? true
```
Comment (laikq) [inline pkgs/applications/graphics/farge/default.nix:1]: You don't need config and pkgs as function arguments here. config isn't used at all, and instead of pkgs, you can directly have bash, imagemagick, slurp, grim, wl-clipboard and feh as arguments.

File: pkgs/applications/graphics/farge/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "farge";
+  version = "1.0.8";
+  src = fetchFromGitHub {
+    owner = "sdushantha";
+    repo = "farge";
+    rev = "f8fa80f7bebc174ab38efb02292b4d0dec03a7dc";
+    sha256 = "08z7dgc3wygl6s922s4gsi2fkrfmghzy82ivnivxr293b9mll7ar";
+  };
+
+  propagatedBuildInputs = with pkgs;
```
Comment (laikq) [inline pkgs/applications/graphics/farge/default.nix:14]: Also, is there a reason this isn't `buildInputs`? I don't see why these inputs should be propagated...

Comment (loafofpiecrust) [reply pkgs/applications/graphics/farge/default.nix:14]: I added them as propagated because farge needs these programs on path at runtime, and my impression of the difference is that `buildInputs` doesn't do this. Maybe I misunderstand the difference?

**Comment by loafofpiecrust** [comment]
@laikq Thank you for the feedback! I suspected it wouldn't work on X11 as it does require `colorpicker` there. I've only been using this on Wayland. It seems like `colorpicker` is not already in `nixpkgs`, and I don't have motivation to write that up too. Is that a deal breaker for initial inclusion here?

File: pkgs/applications/graphics/farge/default.nix (hunk 2 above)
Comment (laikq) [reply pkgs/applications/graphics/farge/default.nix:14]: I think the main difference between propagatedBuildInputs and buildInputs is that: If a derivation A has a derivation B in its buildInputs, and derivation B has some derivations C, D, E in its propagatedBuildInputs, then A can see B, C, D and E in its `PATH` variable at build-time (and therefore A can, for example, save the path to C, D and E and execute them later at run-time). If C, D, and E are in the buildInputs (instead of propagatedBuildInputs) of B, then A only has B in its `PATH` at run-time. I hope someone corrects me if I'm wrong :upside_down_face: 

But, I'm fairly certain that Nix doesn't "sandbox" any executables by default. What I mean is that the environment in which farge runs is the same as the environment from which it is executed, so `PATH` isn't changed by Nix, and `farge` can't find its buildInputs at run-time. But, this can be fixed easily, using the `makeWrapper` and `wrapProgram` shell functions from stdenv. I always use [ripgrep](https://github.com/BurntSushi/ripgrep) to find out real-world examples of how to use NIx functions, and it seems you can just do something like this: `wrapProgram $out/bin/farge --prefix PATH : ${stdenv.lib.makeBinPath buildInputs}` (in `postInstall` for example) to make all buildInputs visible to farge via the `PATH` variable at run-time.

Now, I wonder how you got the packages available at run-time, though. Could it be that you already have wl-clipboard, slurp and grim installed in your user environment, maybe? Or ran farge from `nix-shell`? If not, I have probably misunderstood something.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -4659,6 +4659,12 @@
       fingerprint = "74F5 E5CC 19D3 B5CB 608F  6124 68FF 81E6 A785 0F49";
     }];
   };
+  loafofpiecrust = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:4662]: IIRC, adding yourself as a maintainer is preferred to be a separate commit.

Something like, `maintainers: add loafofpiecrust`. Then apply the farge commit.

File: pkgs/applications/graphics/farge/default.nix (hunk 2 above)
Comment (Infinisil) [reply pkgs/applications/graphics/farge/default.nix]: @laikq You're very much on point with your explanation, nice!

File: pkgs/applications/graphics/farge/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  makeFlags = [ "DEST=$(out)/bin" ];
+  preInstall = ''
+    mkdir -p $out/bin
+  '';
+  postInstall = ''
+    wrapProgram "$out/bin/farge" --prefix PATH : "${lib.makeBinPath buildInputs}"
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/farge/default.nix:27]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/graphics/farge/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{ lib, makeWrapper, fetchFromGitHub, bash, imagemagick, slurp, grim
+, wl-clipboard, feh, stdenv, waylandSupport ? true, imageSupport ? true }:
+
+stdenv.mkDerivation rec {
+  pname = "farge";
+  version = "1.0.8";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/farge/default.nix:7]:
```suggestion
  version = "1.0.8";

  src = fetchFromGitHub {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.
